kernel/init/init.c

```diff
@@ -0,0 +1,53 @@
+//! @file init.c
+//! @brief File containing kernel entrypoint and kernel init functions
+
+#include <init/stivale2.h> // For stivale2 protocol type definitions
+#include <lib/log.h>	   // For log_printf
+#include <sys/ports.h>
+
+//! @brief Stack used by kernel on bootstrap
+//! As dictated by stivale2 protocol, we should provide an address of the stack
+//! that we will use to the bootloader. As bootloader needs to know the stack
+//! before kernel even starts, we allocate stack statically as a global
+//! uninitialized array
+static uint8_t kernel_stack[4096];
+
+// This part is taken from limine-bootloader/limine-barebones
+// (https://github.com/limine-bootloader/limine-barebones/blob/master/src-stivale2/kernel.c)
+// The stivale2 specification says we need to define a "header structure".
+// This structure needs to reside in the .stivale2hdr ELF section in order
+// for the bootloader to find it. We use this __attribute__ directive to
+// tell the compiler to put the following structure in said section.
+__attribute__((section(".stivale2hdr"),
```

Comment:
Indentation's weird here imo

Reply:
I don't care about it that much, that's just what clang-format does, and idk how to modify it to not do that